library/common/jni/jni_interface.cc

```diff
@@ -271,18 +279,27 @@ static envoy_filter_data_status jvm_http_filter_on_request_data(envoy_data data,
   env->DeleteLocalRef(j_data);
 
   return (envoy_filter_data_status){/*status*/ unboxed_status,
-                                    /*data*/ native_data};
+                                    /*data*/ native_data,
+                                    /*pending_headers*/ pending_headers};
 }
 
 static envoy_filter_data_status jvm_http_filter_on_response_data(envoy_data data, bool end_stream,
                                                                  const void* context) {
   JNIEnv* env = get_env();
   jobjectArray result = static_cast<jobjectArray>(
       jvm_on_data("onResponseData", data, end_stream, const_cast<void*>(context)));
+  jsize size = env->GetArrayLength(result);
 
   jobject status = env->GetObjectArrayElement(result, 0);
   jobject j_data = static_cast<jobjectArray>(env->GetObjectArrayElement(result, 1));
 
+  envoy_headers* pending_headers = nullptr;
```

Comment:
Can we create static funcs for the repeated code in these functions?

Reply:
I think the common factoring that's already here complicates following the code in this file somewhat, and would hesitate to push it further.

Reply:
Not sure I completely agree. But feel free to TIOLI!

Reply:
Thanks, I do appreciate the perspective. I'll think further about how this could be simplified, but for now I'm going to stick with my inclination to leave it. Note there's a great deal of dispatch to common-factored paths already occurring. e.g. the jvm_http_filter_on_response_headers delegates to jvm_on_response_headers, which delegates pass_headers, all of which make use of further extracted common utility functions. While this common factoring can serve to encapsulate logic, it also means you need to jump between more locations to follow what's actually happening.